pages/implementation/start-using-avo-codegen.mdx

```diff
@@ -5,15 +5,59 @@ title: 'Start using Avo Codegen'
 
 import Link from '../../components/Link';
 
-In this guide we'll cover all the required steps to start using Avo in your app or website, both alongside existing tracking code and SDKs and in a new project.
+## What is Codegen
+
+Codegen produces type safe code for implementing analytics. 
+
+- The "data designer" specifies the event structure in the <Link passHref href="/workspace"><a>Avo Tracking Plan</a></Link>, and which platforms should send the event
+- then the developer who implements analytics can use Codegen to implement the analytics calls per each analytics event
+
+## Why use Codegen
+
+Codegen simplifies implementation and makes sure it is correct and consistent across all platforms.
+
+For example, instead of Android calling `analytics.track("game started")` and iOS calling `analytics.track("gameStarted")` , they both call `Avo.gameStarted()`, and the Avo function takes care of the spelling of the event and property names as they get passed into the analytics SDK.
+
+In other words, event and property naming is abstracted entirely away from the event implementation layer, ensuring that events and properties are named the same across all platforms, teams and code bases.
+
+## When use Codegen
+
+Codegen is our take on increasing the robustness of the analytics implementation and simplifying the implementation process.
+
+Adopting Codegen is not required to get value from Avo - many companies have their own ways to achieve robust analytics implementation and prefer to stick to them after adopting Avo and it’s totally fine.
+
+Avo is designed to be efficient and powerful without the Codegen, providing the option of the advanced tracking plan management with any implementation approach. 
+The decision to use Codegen is usually in the hands of the Engineering department. If they prefer to use a different approach Avo would help you communicate the tracking plan changes to the engineering team and help them to give the implementation feedback.
```

Comment:
Want to avoid using "them" as a term for engineers here. I think we should be writing this doc for developers, who perhaps received a link from a data person.
```suggestion
The decision to use Codegen is usually in the hands of the engineering team on each platform. If you prefer to use a different approach for implementing your tracking code Avo can still help you by providing a platform to collaborate and align on the tracking specs, and by providing clear specs for the suggested tracking changes.
```